src/types/Message.ts

```diff
@@ -0,0 +1,184 @@
+import type { ICredential, ICredentialPresentation, IAttestation, CTypeHash, DidResourceUri, DidUri, IDelegationNode,PartialClaim } from '@kiltprotocol/types'
+import type { IQuoteAgreement } from './Quote'
+
+
+export interface IDelegationData {
+  account: IDelegationNode['account']
+  id: IDelegationNode['id']
+  parentId: IDelegationNode['id']
+  permissions: IDelegationNode['permissions']
+  isPCR: boolean
+}
+
+export type MessageBodyType =
+  | 'error'
+  | 'reject'
+  | 'submit-terms'
+  | 'request-attestation'
+  | 'request-payment'
+  | 'confirm-payment'
+  | 'submit-attestation'
+  | 'reject-attestation'
+  | 'request-credential'
+  | 'submit-credential'
+  | 'reject-terms'
+
+interface IMessageBodyBase {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  content: any
+  type: MessageBodyType
+}
+
+export interface IError extends IMessageBodyBase {
+  content: {
+    /** Optional machine-readable type of the error. */
+    name?: string
+    /** Optional human-readable description of the error. */
+    message?: string
+  }
+  type: 'error'
+}
+
+export interface IReject extends IMessageBodyBase {
+  content: {
+    /** Optional machine-readable type of the rejection. */
+    name?: string
+    /** Optional human-readable description of the rejection. */
+    message?: string
+  }
+  type: 'reject'
+}
+
+export interface ISubmitTerms extends IMessageBodyBase {
+  content: ITerms
+  type: 'submit-terms'
+}
+
+export interface IRequestAttestationContent {
+  credential: ICredential
+  quote?: IQuoteAgreement
+}
+
+export interface IRequestAttestation extends IMessageBodyBase {
+  content: IRequestAttestationContent
+  type: 'request-attestation'
+}
+
+export interface IRequestPaymentContent {
+  // Same as the `rootHash` value of the `'request-attestation'` message */
+  claimHash: string
+}
+
+export interface IRequestPayment extends IMessageBodyBase {
+  content: IRequestPaymentContent
+  type: 'request-payment'
+}
+
+export interface ISubmitAttestationContent {
+  attestation: IAttestation
+}
+
+export interface ISubmitAttestation extends IMessageBodyBase {
+  content: ISubmitAttestationContent
+  type: 'submit-attestation'
+}
+
+export interface IRejectAttestation extends IMessageBodyBase {
+  content: ICredential['rootHash']
+  type: 'reject-attestation'
+}
+
+export interface IRequestCredentialContent {
+  cTypes: Array<{
+    cTypeHash: CTypeHash
+    trustedAttesters?: DidUri[]
+    requiredProperties?: string[]
+  }>
+  challenge?: string
+}
+
+export interface IRejectTermsContent {
+  claim: PartialClaim
+  legitimations: ICredential[]
+  delegationId?: IDelegationNode['id']
+}
+
+
+
+export interface ISubmitCredential extends IMessageBodyBase {
+  content: ICredentialPresentation[]
+  type: 'submit-credential'
+}
+
+export interface IRequestCredential extends IMessageBodyBase {
+  content: IRequestCredentialContent
+  type: 'request-credential'
+}
+
+export interface IConfirmPaymentContent {
```

Comment:
this should be documented

Reply:
I hope this is fine: [3f3aa01](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/3f3aa01f2af6dbbd023d75da81ff8442eebb1501)